pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -362,12 +362,14 @@ private void writeMetadata()
     properties.setProperty(DIMENSIONS, config.getDimensions());
     properties.setProperty(METRICS, config.getMetrics());
     properties.setProperty(DATETIME_COLUMNS, config.getDateTimeColumnNames());
-    properties.setProperty(TIME_COLUMN_NAME, config.getTimeColumnName());
+    String timeColumnName = config.getTimeColumnName();
+    if (timeColumnName != null && !config.getSchema().getDateTimeNames().contains(timeColumnName)) {
```

Comment:
Why do we need the second part of the check? What if the schema has no DateTimeField but only TimeField?

Reply:
Summarizing our offline discussion:
The second part was to avoid setting the time column name in `segment.datetime.column.names` and also in `segment.time.column.name`, which will happen if the primary time column is **dateTimeFieldSpec**. This happens because `segment.time.column.name` is **serving as primary timeColumnName and also as timeFieldSpec.** This worked well so far, because primary time column and timeFieldSpec were one and the same. 
But based on our chat, removing that check. As decided, the `segment.time.column.name` property will serve as the primary time column name. It will match with timecolumnName in tableConfig. It can be of type DATE_TIME or TIME.